eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs

```diff
@@ -296,13 +304,66 @@ instance EraPParams era => EncCBOR (ProposalProcedure era) where
         !> To pProcGovernanceAction
         !> To pProcAnchor
 
+data Constitution era = Constitution
+  { constitutionHash :: !(SafeHash (EraCrypto era) ByteString)
```

Comment:
Because we never do the actual hashing or verification we can set the phantom type to whatever we like (even if it is recursive like that). And ByteString is a lot worse than `Constitution`
```suggestion
  { constitutionHash :: !(SafeHash (EraCrypto era) Constitution)
```

Reply:
TIL! Thanks Alexey! :raised_hands: